### DISCUSSION TITLE
Leaf server returned an error

### DISCUSSION THREAD

**Discussion by Sreeharika22** [uwrit/leaf#577] [Q&A]
Hi, 

I am trying to run leaf server as localhost, but when I run leaf API & perform npm start server http://localhost:3000/ opens up but returns with error. my visual code returns following set of errors.

![Local host 3000 error](https://user-images.githubusercontent.com/115589971/200433405-929d34d1-a46f-4133-b1dd-be14e88ae276.png)
![.NET package error](https://user-images.githubusercontent.com/115589971/200431473-258eb7d5-cfec-4ee9-b7de-5637cbab83cf.png)
![Visual Code UI errors](https://user-images.githubusercontent.com/115589971/200431656-1f7c0455-38a1-42c2-bd90-e23186ab6dfb.png)

Please let me know how to resolve this!

Also, which SQL software to use on macOS for database? I know, SSMS is recommended for leaf but SSMS is not available for macOS. what is the best other option to work with databases?

Thanks,
Harika.

**Comment by ndobb** [comment]
Hey @Sreeharika22 - 
1. In the browser, can you go the Network tab and highlight which API calls are failing?
2. Are you running the API in a Docker container or debugging via Visual Studio?
3. What do the Leaf API logs (in `SERILOG_DIR` environment variable) indicate?

For SQL on macOS, we use Azure Data Studio.
